text/0000-refactor-sbom.md

```diff
@@ -0,0 +1,496 @@
+
+**OCI v1.1** - pending new releases to the OCI image and distribution specs, as described [here](https://opencontainers.org/posts/blog/2023-07-07-summary-of-upcoming-changes-in-oci-image-and-distribution-specs-v-1-1/)
+
+Notably, these new specs codify ways of "attaching" arbitrary OCI artifacts to an image -
+ - via the Referrers API if supported by the registry
+   - the manifest for each artifact should contain:
+     - `subject` to point to the image described by the artifact
+ - via Referrers Tag Schema (fallback) if Referrers API is not supported by the registry
+   - a manifest "list" published to tag `sha256-<image digest>` where each list element should contain:
+     - `artifactType` corresponding to the `config.mediaType` in the manifest
+
+# Motivation
+[motivation]: #motivation
+
+- Why should we do this? Our current approach has a few drawbacks, namely:
+  * It can make application images quite large
+  * It can be hard to find SBOMs for buildpacks-built images; ecosystem tooling such as `cosign download sbom` won't work (see [PR 278](https://github.com/buildpacks/rfcs/pull/278))
+  * There is currently no clear way to associate SBOMs for build and run base images with an application image (today this is unspec'd and entirely up to the platform to manage)
+  
+- What use cases does it support?
+  * As a buildpacks user, I want **signed** SBOMs for my images in an easily consumable format.
+  * ...
+
+- What is the expected outcome?
+  * Smaller application images
+  * Integration with ecosystem tooling
+  * (Eventually) A more complete SBOM for buildpacks-built images
+
+# What it is
+[what-it-is]: #what-it-is
+
+See [How it Works](#how-it-works).
+
+# How it Works
+[how-it-works]: #how-it-works
+
+## Setup
+
+The lifecycle will ship with a new `signer` binary.
+In the past we've talked about making this binary totally separate from the lifecycle,
+but given the need to restore SBOM data and thus teach the lifecycle about attestations, there isn't much value there.
+We should make it possible to invoke the `signer` separately from other phases.
+It should not be necessary to run the `signer` within the build container (although it might be more convenient to run it that way).
+
+Buildpacks should continue to output SBOM files as before. We won't need to bother buildpack authors with this change.
+
+## Export
+
+The `exporter` will accept a new optional flag `-omit-sbom` that defaults to `false`.
+* If `-omit-sbom=false` the exporter will behave as today, and include the SBOM layer containing the `<layers>/sbom/launch/` directory in the application image.
+* If `-omit-sbom=true` the exporter will not create the SBOM layer, but `<layers>/sbom/launch/` will remain on the filesystem and be available to the `signer`.
+
+The exporter will additionally generate configuration for the `signer` irrespective of the value of `-omit-sbom`.
+
+## Sign
```

Comment:
From 10/5 Working Group - adding signatures could be a big jump for platforms that just want to get the SBOM out of the application image. We could provide an option to generate **unsigned** attestations as OCI artifacts in the registry which might ease the migration path for some folks.